### DISCUSSION TITLE
Contributing

### DISCUSSION THREAD

**Discussion by popkinj** [maplibre/martin#1170] [General]
Hey Martin Team.

I've been using Martin for a couple of new projects. Really like it so far. :)

I'd like to check in on "the vibe". The contributions section of the README is rather sparse. Are PRs welcome? If so, is there a roadmap for new features and such? For example, I'd really like to add a [filter URL parameter](https://github.com/CrunchyData/pg_tileserv/blob/master/hugo/content/usage/cql.md), kinda like how pg_tileserv has. Is that something that would be welcome?

**Comment by nyurik** [comment]
Hi @popkinj , we have a fairly active community without a well defined roadmap - mostly because there is no full time developers dedicated to this project, and so it develops by natural growth - if someone wants a specific feature - lets create an issue, discuss it, approve it, someone implements it, and we merge. Simple, eh? :)

With regards to the filter URL, that one is rather tricky. Martin [PG function sources](https://maplibre.org/martin/sources-pg-functions.html) allow an arbitrary set of query values to be passed into your own PG function - so it would be fairly easy for you to implement any kind of filtering directly in an SQL function. I am not as eager for arbitrary filtering for everyone because this is a relatively small usecase, but at the same time it allows any installation to have significant unexpected performance cost.  I might be wrong of course, and we should discuss it, but there has to be a really compelling case for this that cannot be solved with the custom functions.

**Comment by popkinj** [comment]
Thanks @nyurik . You make a very good case. 

Before I hastily create a new ticket... Maybe I should test out the PG function route to see how flexible it is. My original thinking was to have drop-in replacement for pg_tileserve, which is what I was using previously. But there's nothing wrong with doing things differently. :)

**Reply by nyurik** [reply]
sure thing... afterall, if one programming language was a drop in for another one, we would still be using Cobol+++ ;)
